src/System.Text.JsonLab.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -28,7 +28,7 @@ public static JsonDynamicObject Parse(ReadOnlySpan<byte> utf8, int expectedNumbe
             var properties = new Dictionary<JsonProperty, JsonValue>(expectedNumberOfProperties);
             stack.Push(new JsonDynamicObject(properties));
 
-            var reader = new Utf8JsonReader(utf8);
```

Comment:
I was late to the review. Where can I find some writeup on the motivation for the name change?

Reply:
https://github.com/dotnet/apireviews/pull/73

Alternatives consider (either of which I am fine with): `Utf8JsonReader`, `JsonReaderUtf8`, `JsonReader`

What name would you prefer?

Reply:
It seems like we have Utf8Parser, Utf8Formatter, and so it seems Utf8JsonReader would follow the pattern.

Reply:
> But I was mainly asking about the reasons.

- We want all JsonX to sort together. JsonReaderState, JsonUtf8Reader, JsonReaderOptions, JsonUtf8Writer, etc.
- The feature is JSON, not UTF-8, where UTF-8 is a characteristic of the reader. Putting Utf8 up-front emphasizes the incorrect aspect of the type. Arguable, Utf8Parser/Formatter are primarily about UTF-8.

Reply:
> we have Utf8Parser, Utf8Formatter

Arguably this is a Utf8Reader.  What kind?  A JsonUtf8Reader.

But I don't actually have a strong opinion.